templates/php-component/docker/php-prod.ini

```diff
@@ -0,0 +1,18 @@
+; Recommended production values
```

Comment:
Nejsou production default v tom base image? Naopak bych se bál, že nám to časem vytimeoutuje

Reply:
kupodivu nejsou
co vytimeoutuje?

Reply:
No, třeba že časem bude default jiné pořadí GPCS parametrů. Nebo tam zůstanou jen GP... a nám to zůstane špatně.

Reply:
To bolo vyextrahovane z ofiko php.ini. Myslim, ze by malo stacit, ze to raz za cas (napr. pri prechode na dalsiu major verziu) skontolujeme.

Reply:
menit by se to mohlo teoreticky snad jen pri zmene FROM php:7

Reply:
jo, tiez si myslim. v ofiko phpcku to maju normalne ako dva subory a to co mame je myslim len diff.
https://github.com/php/php-src/tree/PHP-7.1 (a tam je https://github.com/php/php-src/blob/PHP-7.1/php.ini-development a https://github.com/php/php-src/blob/PHP-7.1/php.ini-production)

```diff
--- php.ini-development.txt	2018-03-07 17:48:00.015929208 +0100
+++ php.ini-production.txt	2018-03-07 17:48:11.763820374 +0100
@@ -83,7 +83,7 @@
 ; development version only in development environments, as errors shown to
 ; application users can inadvertently leak otherwise secure information.
 
-; This is php.ini-development INI file.
+; This is php.ini-production INI file.
 
 ;;;;;;;;;;;;;;;;;;;
 ; Quick Reference ;
@@ -457,7 +457,7 @@
 ; Development Value: E_ALL
 ; Production Value: E_ALL & ~E_DEPRECATED & ~E_STRICT
 ; http://php.net/error-reporting
-error_reporting = E_ALL
+error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT
 
 ; This directive controls whether or not and where PHP will output errors,
 ; notices and warnings too. Error output is very useful during development, but
@@ -474,7 +474,7 @@
 ; Development Value: On
 ; Production Value: Off
 ; http://php.net/display-errors
-display_errors = On
+display_errors = Off
 
 ; The display of errors which occur during PHP's startup sequence are handled
 ; separately from display_errors. PHP's default behavior is to suppress those
@@ -485,7 +485,7 @@
 ; Development Value: On
 ; Production Value: Off
 ; http://php.net/display-startup-errors
-display_startup_errors = On
+display_startup_errors = Off
 
 ; Besides displaying errors, PHP can also log errors to locations such as a
 ; server-specific log, STDERR, or a location specified by the error_log
@@ -529,7 +529,7 @@
 ; Development Value: On
 ; Production Value: Off
 ; http://php.net/track-errors
-track_errors = On
+track_errors = Off
 
 ; Turn off normal error reporting and emit XML-RPC error XML
 ; http://php.net/xmlrpc-errors
@@ -1197,7 +1197,7 @@
 ; Enable / Disable collection of memory usage statistics by mysqlnd which can be
 ; used to tune and monitor MySQL operations.
 ; http://php.net/mysqlnd.collect_memory_statistics
-mysqlnd.collect_memory_statistics = On
+mysqlnd.collect_memory_statistics = Off
 
 ; Records communication from all extensions using mysqlnd to the specified log
 ; file.
@@ -1575,7 +1575,7 @@
 ; Development Value: 1
 ; Production Value: -1
 ; http://php.net/zend.assertions
-zend.assertions = 1
+zend.assertions = -1
 
 ; Assert(expr); active by default.
 ; http://php.net/assert.active
@@ -1897,7 +1897,7 @@
 
 ; Enables or disables copying of PHP code (text segment) into HUGE PAGES.
 ; This should improve performance, but requires appropriate OS configuration.
-;opcache.huge_code_pages=0
+;opcache.huge_code_pages=1
 
 ; Validate cached file permissions.
 ;opcache.validate_permission=0
```

ale pozeram, ze by sa to asi dalo malicko zjednodusit

Reply:
Mě už se to jednou právě vymstilo, když jsem měl verzovaný celý `php.ini`. Bylo to okolo té doby, co se [měnil owner u toho socketu](https://bugs.php.net/bug.php?id=67244). Ale nemám na to silný názor, spíš slabý :D